extensions/purdue-course-search/src/index.tsx

```diff
@@ -0,0 +1,96 @@
+import { useCallback, useState } from "react";
+
+import { ActionPanel, Action, List } from "@raycast/api";
+import { useStreamJSON } from "@raycast/utils";
+
+type SearchResultScheduleType = "Lecture" | "Distance Learning" | "Experiential" | "Laboratory" | "Individual Study";
+
+type SearchResult = {
+  id: string;
+  title: string;
+  subjectCode: string;
+  courseCode: string;
+  instructor: string[];
+  description: string;
+  capacity: number;
+  credits: [number] | [number, number];
+  term: string;
+  crn: number[];
+  sched: SearchResultScheduleType[];
+};
+
+export default function Command() {
+  const [searchText, setSearchText] = useState("");
+
+  const searchResultsFilter = useCallback(
+    (searchResult: SearchResult) => {
+      if (!searchText) {
+        return true;
+      }
+
+      const lowerSearchText = searchText.toLowerCase();
+      return (
+        searchResult.title.toLowerCase().includes(lowerSearchText) ||
+        searchResult.subjectCode.toLowerCase().includes(lowerSearchText) ||
+        searchResult.courseCode.toLowerCase().includes(lowerSearchText)
+      );
+    },
+    [searchText],
+  );
+
+  const searchResultsTransform = useCallback((item: SearchResult): SearchResult => {
+    return {
+      ...item,
+      id: `${item.subjectCode}${item.courseCode}`,
+    };
+  }, []);
+
+  const { data, isLoading } = useStreamJSON(
+    "https://boilerclasses.s3.us-east-2.amazonaws.com/data/classes_fall2024.json",
```

Comment:
Wouldn't this S3 bucket keep getting updated every quarter? It would not be extensible if you have to update this extension per quarter to reflect correct classes. If they keep consistent object key pattern of `classes_{fall|winter|spring|summer}{year}.json`, then I'd recommend generating the quarter and year client-side and then querying the S3 object.